src/utils/ensure-base-exists.utils.ts

```diff
@@ -39,54 +39,65 @@ export const ensureBaseTestsExists = async ({
   base: string | null;
   context: Context;
   octokit: InstanceType<typeof GitHub>;
-}): Promise<{ currentBaseSha: string } | null> => {
+}): Promise<{ shaToCompareAgainst: string | null }> => {
   const logger = log.getLogger(METICULOUS_LOGGER_NAME);
 
   // Running missing tests on base is only supported for Pull Request events
   if (event.type !== "pull_request" || !base) {
-    return null;
+    return { shaToCompareAgainst: null };
   }
 
   const { owner, repo } = context.repo;
   const baseRef = event.payload.pull_request.base.ref;
 
+  logger.debug(JSON.stringify({ base, baseRef }, null, 2));
+
+  const testRun = await getLatestTestRunResults({
+    client: createClient({ apiToken }),
+    commitSha: base,
+  });
+
+  if (testRun != null) {
+    logger.log(`Tests already exist for commit ${base} (${testRun.id})`);
+    return { shaToCompareAgainst: base };
+  }
+
+  const { workflowId } = await getCurrentWorkflowId({ context, octokit });
+
+  // We can only trigger a workflow_run against the head of the base branch
+  // This will give some spurious diffs if it's different from `base`, but it's the best we can do
+
   const currentBaseSha = await getHeadCommitForRef({
     owner,
     repo,
     ref: baseRef,
     octokit,
   });
 
-  logger.debug(JSON.stringify({ base, baseRef, currentBaseSha }, null, 2));
+  logger.debug(
+    JSON.stringify({ owner, repo, base, baseRef, currentBaseSha }, null, 2)
+  );
   if (base !== currentBaseSha) {
-    const message = `Pull request event received ${base} as the base commit but ${baseRef} \
-is now pointing to ${currentBaseSha}. Will use ${currentBaseSha} for Meticulous tests.`;
+    const message = `Meticulous tests on base commit ${base} have either not completed or never ran so we have nothing to compare against.
+    In addition we were not able to trigger a run on ${base} since the '${baseRef}' branch is now pointing to ${currentBaseSha}.
+    Therefore no diffs will be reported for this run. Re-running the tests may fix this.`;
     logger.warn(message);
     ghWarning(message);
+    return { shaToCompareAgainst: null };
```

Comment:
I changed this to not do any comparisons in this case: better to show no diffs than to show confusing diffs that include changes from other PRs.